packages/realm-web/src/transports/AuthenticatedTransport.ts

```diff
@@ -0,0 +1,48 @@
+import { Transport, Request } from "./Transport";
+import { PrefixedTransport } from "./PrefixedTransport";
+
+interface UserContext {
+    /** The currently active user */
+    currentUser: Realm.User | null;
+}
+
+/**
+ * Fetches resources as a particular user.
+ */
+export class AuthenticatedTransport implements Transport {
+    private readonly userContext: UserContext;
+    private readonly transport: Transport;
+
+    public constructor(userContext: UserContext, transport: Transport) {
```

Comment:
transport should be the first argument to be on par with the other transports and as the most important argument of the AuthenticatedTransport class